dht/i_partitioner.cc

```diff
@@ -295,43 +295,43 @@ bool ring_position::less_compare(const schema& s, const ring_position& other) co
     return tri_compare(s, other) < 0;
 }
 
-int ring_position_tri_compare(const schema& s, ring_position_view lh, ring_position_view rh) {
+std::strong_ordering ring_position_tri_compare(const schema& s, ring_position_view lh, ring_position_view rh) {
     auto token_cmp = tri_compare(*lh._token, *rh._token);
     if (token_cmp != 0) {
-        return token_cmp < 0 ? -1 : 1;
+        return token_cmp;
     }
     if (lh._key && rh._key) {
         auto c = lh._key->legacy_tri_compare(s, *rh._key);
         if (c) {
-            return c;
+            return c <=> 0;
```

Comment:
Wouldn't "return c" do exactly the same? Or did you deliberately not convert legacy_tri_convert() and c is an integer?

Reply:
Yes, it is an integer.